phases/19-capstone-projects/39-instruction-tuning-sft/code/tests/test_main.py

```
@@ -0,0 +1,174 @@
+    def test_decode_response_drops_specials(self) -> None:
+        tok = InstructionTokenizer()
+        ids = [InstructionTokenizer.RESP_ID, ord("h"), ord("i")]
+        self.assertEqual(tok.decode_response(ids), "hi")
+
+
+class CollateTests(unittest.TestCase):
+    def test_collate_pads_and_masks_instruction(self) -> None:
+        tok = InstructionTokenizer()
+        ids1, rs1 = tok.encode_pair("ab", "cd", max_len=32)
+        ids2, rs2 = tok.encode_pair("a", "bcdefg", max_len=32)
+        input_ids, labels, attn_mask = sft_collate([(ids1, rs1), (ids2, rs2)])
```

Ruff (0.15.14)

[warning] Unpacked variable `attn_mask` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)